wombat/src/main/include/behaviour/Trigger.h

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2023-2024 CurtinFRC
+// Open Source Software, you can modify it according to the terms
+// of the MIT License at the root of this project
+
+#pragma once
+
+#include <units/time.h>
+
+#include <functional>
+#include <string>
+
+#include "behaviour/Behaviour.h"
+
+namespace behaviour {
+/**
+ * @breif A class which binds behaviours to boolean conditions. Most commonly used through a wrapper around
+ * a controller.
+ *
+ * Trigger is an API within the Behaviour scheduling to run different Behaviours based off boolean conditions.
+ * It is instantiated with a condition that is evaluated on every Tick of the BehaviourScheduler causing the
+ * OnTick method to run, checking the condition and running the relevant behaviour. Using the Trigger API
+ * provides a variety of advantages to passing controllers into functions. By creating a Trigger using each
+ * controller as a trigger based controller we can easily figure out the control scheme of the entire robot.
```

Comment:
add a comma between "trigger based controller" and "we can easily" for easier reading